MatrixSDKExtensions/VoIP/Jingle/MXJingleCallStackCall.m

```diff
@@ -625,36 +602,32 @@ - (void)setCameraPosition:(AVCaptureDevicePosition)theCameraPosition
 
 #pragma mark - Private methods
 
+//  Not used for now, will be in future
 - (BOOL)isHoldOffer:(NSString *)sdpOffer
 {
-    NSString *keyword = @"m=audio";
-    NSError *error = NULL;
-    NSRegularExpression *regex = [NSRegularExpression regularExpressionWithPattern:keyword
-                                                                           options:NSRegularExpressionCaseInsensitive
-                                                                             error:&error];
-    NSUInteger numberOfAudioTracks = [regex numberOfMatchesInString:sdpOffer options:0 range:NSMakeRange(0, [sdpOffer length])];
-    
-    keyword = @"m=video";
-    error = NULL;
-    regex = [NSRegularExpression regularExpressionWithPattern:keyword
-                                                      options:NSRegularExpressionCaseInsensitive
-                                                        error:&error];
-    NSUInteger numberOfVideoTracks = [regex numberOfMatchesInString:sdpOffer options:0 range:NSMakeRange(0, [sdpOffer length])];
-    
+    NSUInteger numberOfAudioTracks = [self numberOfMatchesOfKeyword:@"m=audio" inString:sdpOffer];
+    NSUInteger numberOfVideoTracks = [self numberOfMatchesOfKeyword:@"m=video" inString:sdpOffer];
+
     if (numberOfAudioTracks == 0 && numberOfVideoTracks == 0)
     {
         //  no audio or video tracks
         return YES;
     }
+
+    NSUInteger numberOfInactiveTracks = [self numberOfMatchesOfKeyword:@"a=inactive" inString:sdpOffer];
```

Comment:
I still think we should be able to get this without sdp parsing by setting the remote description and then querying the tracks, as per https://github.com/matrix-org/matrix-js-sdk/blob/develop/src/webrtc/call.ts#L1025 / https://github.com/matrix-org/matrix-js-sdk/blob/46cebcd1ca51a6dacd9802f9aa3593cef2015436/src/webrtc/call.ts#L690

Reply:
It's not used right now, just didn't delete for possible future uses.